app-backend/tool/src/main/scala/ast/MapAlgebraAST.scala

```diff
@@ -43,49 +46,59 @@ object MapAlgebraAST {
   }
 
   /** Operations which should only have one argument. */
-  abstract class UnaryOp(override val symbol: String) extends Operation(symbol) with Serializable
+  sealed trait UnaryOp extends Operation with Serializable
```

Comment:
It was required to keep pattern matches elsewhere air-tight. As an abstract class, pattern matches non-exhaustion warnings would be fired like: `no branch for case Operation()` which could otherwise never be possible.